File: solvers/Soln.py
import timeit
import json

with open('../questions/add.json', 'r') as f:
    question = json.loads(f.read())

LARGE_QUESTION = question.get("large_question")
QUESTIONS = question.get("questions")


class Soln:
    def solve(self, *args):
        raise NotImplemented

    def run(self):
        for q in QUESTIONS:
            answer = q.get("answer")
            params = q.get("input")
            if (ans := self.solve(*params)) != answer:
                print(f"FAILED:\n{params}\n{ans} != {answer}")
                return False
        else:
            print("SUCCESS")
            return True

    def time(self):
        params = LARGE_QUESTION.get("input")
        answer = LARGE_QUESTION.get("answer")
        if self.solve(*params) == answer:
            print(timeit.timeit(lambda: self.solve(*params), number=10000))


class UserSolution(Soln):
    def solve(self, a, b):
        return a + b


soln = UserSolution()
if soln.run():
    soln.time()

File: solvers/Soln.js
var fs = require('fs');

class Soln {
  constructor() {
    let rawdata = fs.readFileSync('./question.json');
    const question = JSON.parse(rawdata);
    this.questions = question.questions
    this.largeQuestion = question["large_question"]
  }

  solve() {}

  run() {
    for (let question of this.questions) {
      let ans = this.solve(...question.input)
      if(ans !== question.answer) {
        console.log("FAILED\n", question.input, '\n', question.answer, '!=',ans)
        return false
      }
    }
    console.log("SUCCESS")
    return true
  }

  time() {
    let startTime = process.hrtime();
    for (let i = 0; i <= 1000; i++){
        this.solve(this.largeQuestion.input)
    }
    let elapsedSeconds = parseHrtimeToSeconds(process.hrtime(startTime));
    console.log(elapsedSeconds / 1000)
  }
}


class UserSolution extends Soln {
  solve(a, b) {
    return a + b
  }
}

const s = new UserSolution
if(s.run()) s.time()
